packages/next/src/build/utils.ts

```diff
@@ -1528,55 +1617,28 @@ export async function isPageStatic({
               ]
             : await collectGenerateParams(tree)
 
-        appConfig = generateParams.reduce(
-          (builtConfig: AppConfig, curGenParams): AppConfig => {
-            const {
-              dynamic,
-              fetchCache,
-              preferredRegion,
-              revalidate: curRevalidate,
-            } = curGenParams?.config || {}
-
-            // TODO: should conflicting configs here throw an error
-            // e.g. if layout defines one region but page defines another
-            if (typeof builtConfig.preferredRegion === 'undefined') {
-              builtConfig.preferredRegion = preferredRegion
-            }
-            if (typeof builtConfig.dynamic === 'undefined') {
-              builtConfig.dynamic = dynamic
-            }
-            if (typeof builtConfig.fetchCache === 'undefined') {
-              builtConfig.fetchCache = fetchCache
-            }
-
-            // any revalidate number overrides false
-            // shorter revalidate overrides longer (initially)
-            if (typeof builtConfig.revalidate === 'undefined') {
-              builtConfig.revalidate = curRevalidate
-            }
-            if (
-              typeof curRevalidate === 'number' &&
-              (typeof builtConfig.revalidate !== 'number' ||
-                curRevalidate < builtConfig.revalidate)
-            ) {
-              builtConfig.revalidate = curRevalidate
-            }
-            return builtConfig
-          },
-          {}
-        )
+        appConfig = resolveAppConfig(generateParams)
 
         if (appConfig.dynamic === 'force-static' && pathIsEdgeRuntime) {
           Log.warn(
             `Page "${page}" is using runtime = 'edge' which is currently incompatible with dynamic = 'force-static'. Please remove either "runtime" or "force-static" for correct behavior`
           )
         }
 
-        // If force dynamic was set and we don't have PPR enabled, then set the
-        // revalidate to 0.
-        // TODO: (PPR) remove this once PPR is enabled by default
-        if (appConfig.dynamic === 'force-dynamic' && !ppr) {
-          appConfig.revalidate = 0
+        // When PPR is enabled and this page is an App Page, then it should be
+        // partially pre-rendered.
+        if (
+          nextConfigPPREnabled &&
+          routeModule.definition.kind === RouteKind.APP_PAGE
+        ) {
+          isPPR = true
+        }
+
+        // When `dynamic = "force-dynamic"`, then we should never prerender
+        // this page statically unless PPR is enabled, in which case the static
+        // shell can always be prerendered.
+        if (appConfig.dynamic === 'force-dynamic') {
+          appConfig.revalidate = isPPR ? false : 0
```

Comment:
When a page is going to be rendered using PPR, if `force-dynamic` is set on the segment config, the entire page at the root will postpone, resulting in a completly static (and safe) payload. This can be cached for lifetime of the build.